profiler/nsys_mine.py

```diff
@@ -178,14 +188,20 @@ def _plot_total_call(fv3_kernel_timings: Dict[str, List[int]]):
 
 def _filter_kernel_name(kernels: List[Any]) -> List[Any]:
     """Filter the gridtools c++ kernel name to a readable name"""
-    # Run a query to convert the stencil generated string to a readable one
+    # Run a regex to convert the stencil generated string to a readable one
     approx_stencil_name_re = re.search(
-        "(?<=bound_functorIN)(.*)(?=___gtcuda)",
+        "(?<=bound_functor)(.*?)(?=_pyext)",
         kernels[KernelReportIndexing.NAME.value],
     )
     if approx_stencil_name_re is None:
         return kernels
-    approx_stencil_name = approx_stencil_name_re.group().lstrip("0123456789 ")
+    # Clean up & insert
+    approx_stencil_name = (
+        approx_stencil_name_re.groups()[0]
+        .lstrip("IN0123456789<")
+        .replace(" ", "")
+        .replace("____gtcuda", "")
```

Comment:
Might be useful to parameterize the backend, could be `gtc:cuda` soon. That can wait of course...

Reply:
Good point but I expect the generated code to differ so much I'll need to parametrize this entire routine